app/src/main/java/agency/five/tmdb/MainScreen.kt

```diff
@@ -0,0 +1,218 @@
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.navigation.NavController
+
+@Composable
+fun MainScreen(
+    navController: NavController
+) {
+
+    // movies
+    val m0 = MovieItemViewState(
+        id = 0,
+        overview = "Nothing to show.",
+        title = "Iron Man",
+        imageResId = R.drawable.iron_man_1
+    )
+
+    val m1 = MovieItemViewState(
+        id = 1,
+        overview = "Nothing to show.",
+        title = "Gattaca",
+        imageResId = R.drawable.gattaca
+    )
+
+    val m2 = MovieItemViewState(
+        id = 2,
+        overview = "Nothing to show.",
+        title = "Lion King",
+        imageResId = R.drawable.lion_king
+    )
+
+    val m3 = MovieItemViewState(
+        id = 3,
+        overview = "Nothing to show.",
+        title = "Puppy Love",
+        imageResId = R.drawable.puppy_love
+    )
+
+    val m4 = MovieItemViewState(
+        id = 4,
+        overview = "Nothing to show.",
+        title = "Godzila",
+        imageResId = R.drawable.godzila
+    )
+
+    val m5 = MovieItemViewState(
+        id = 5,
+        overview = "Nothing to show.",
+        title = "Jungle Beat",
+        imageResId = R.drawable.jungle_beat
+    )
+
+
+    //list of movies to be shown on different sections tabs
+    val l1 = listOf(m0, m1, m2, m3)
+    val l2 = listOf(m0, m1)
+    val l3 = listOf(m4, m5)
+
+    // map of movies for the "What's popular" section with keys as tab names
+    val map1 = mapOf(
+        stringResource(id = R.string.streaming_tab) to l1,
+        stringResource(id = R.string.on_tv_tab) to l2,
+        stringResource(id = R.string.for_rent_tab) to l3,
+        stringResource(id = R.string.in_theaters_tab) to l1
+    )
+
+    // map of movies for the "Free to watch" section with keys as tab names
+    val map2 = mapOf(
+        stringResource(id = R.string.movies_tab) to l2,
+        stringResource(id = R.string.tv_tab) to l2
+    )
+
+    // map of movies for the "Trending" section with keys as tab names
+    val map3 = mapOf(
+        stringResource(id = R.string.today_tab) to l1,
+        stringResource(id = R.string.this_week_tab) to l3
+    )
+
+
+
+    var home by remember { mutableStateOf(true) }
+
+    Scaffold(
+        topBar = { TopAppBar(
+            backgroundColor = Color(0xFF0B253F)
+        ) {
+            Box(
+                modifier = Modifier.fillMaxSize(),
+                contentAlignment = Alignment.Center) {
+
+                Image(
+                    painter = painterResource(id = R.drawable.ic_logo),
+                    contentDescription = null,
+                    modifier = Modifier.size(width = 134.dp, height = 35.dp))
+
+            }
+        } },
+        bottomBar = { BottomNavigation(
```

Comment:
It would be nice to extract TopBar and BottomBar to separate Composable functions to improve readability of this function. You did that nicely with HomeScreen and Favorites screen.